app/(gcforms)/[locale]/(form administration)/form-builder/[id]/edit/components/settings/SettingsPanel.tsx

```diff
@@ -20,20 +20,22 @@ import { safeJSONParse } from "@lib/utils";
 import { useTemplateContext } from "@lib/hooks/form-builder/useTemplateContext";
 import { FormProperties } from "@lib/types";
 import { useFeatureFlags } from "@lib/hooks/useFeatureFlags";
+import { useFormBuilderConfig } from "@lib/hooks/useFormBuilderConfig";
 
 enum DeliveryOption {
   vault = "vault",
   email = "email",
 }
 
-export const SettingsPanel = ({ keyId = false }: { keyId?: string | false }) => {
+export const SettingsPanel = () => {
   const { t, i18n } = useTranslation("form-builder");
   const lang = i18n.language === "en" ? "en" : "fr";
   const { status } = useSession();
 
   const { getFlag } = useFeatureFlags();
+  const { hasApiKeyId: hasKey, apiKeyId } = useFormBuilderConfig();
   const apiAccess = getFlag("apiAccess");
-  const hasApiKey = keyId && apiAccess ? true : false;
+  const hasApiKey = hasKey && apiAccess ? true : false;
```

Comment:
Thinking we should remove the flag check here --- if the user has been able to generate a key the panel needs to operate in API key mode or else we'll run into issues --- i.e. being able to set classifications that shouldn't be possible.

Thinking wrapping the flag around the response delivery radio is likely the only spot we should have the feature flag check.